### PR TITLE
Correct stabilization for VOF

### PR DESCRIPTION
**Pull request by hepap**
# Description of the problem

- The stabilization terms in the NS-VOF assembler did not properly include the density, which causes problem for small scale simulations, such as the one performed for the Laser Powder Bed Fusion process.
- The shock capturing scheme in the VOF needed to be improved.

# Description of the solution

- The implementation of the stabilization is now consistent in terms of units.
- The shock capturing scheme in the VOF is now improved.

# How Has This Been Tested?

- No tests were added and existing tests for the VOF were modified.

- [x] gls_navier_stokes_2d/gls_droplet_marangoni_effect
- [x] gls_navier_stokes_2d/gls_vof_ht_adapt-dissipation
- [x] gls_navier_stokes_2d/gls_vof_peeling
- [x] gls_navier_stokes_2d/gls_vof_tanh_filter
- [x] gls_navier_stokes_2d/gls_vof_wetting-inverted
- [x] gls_navier_stokes_2d/gls_vof_tanh_filter
- [x] gls_navier_stokes_2d/gls_vof_wetting-inverted
- [x] gls_navier_stokes_2d/gls_vof_wetting_mesh-adapt

# Documentation

No change

# Future changes

- Ideally, the complete formulation NS and VOF will be described in a theory guide.

### CODE REVIEW

File: source/solvers/vof_assemblers.cc [hunk 1]
```diff
@@ -195,19 +195,35 @@ VOFAssemblerCore<dim>::assemble_rhs(VOFScratchData<dim> &      scratch_data,
...
-      const double   tolerance = 1e-12;
-      Tensor<1, dim> s         = velocity / (velocity.norm() + tolerance);
+      const double tolerance = 1e-12;
+
+      // We neglect to remove the diffusion aligned with the velocity
+      // as is done in the original article. We re-enable those
+      // terms if artificial diffusion becomes a problem
+      // Tensor<1, dim> s = velocity / (velocity.norm() + 1e-12);
+      // const Tensor<2, dim> k_corr      = (r * s) * outer_product(s,
+      // s);
```
Comment (lpsaavedra) [inline source/solvers/vof_assemblers.cc:219]: Is this re-enabled frequently? If yes, isn't it better to use an if that allows to enable it easily?

**Review by lpsaavedra — COMMENTED** [review]
Not very familiar with the VOF assemblers but everything seems good to me. Just a couple of comments/doubts...

File: source/solvers/vof_assemblers.cc (hunk 1 above)
Comment (blaisb) [reply source/solvers/vof_assemblers.cc:219]: This shouldnt be re-enabled. :).

File: source/solvers/vof_assemblers.cc [hunk 2]
```diff
@@ -73,20 +73,19 @@ VOFAssemblerCore<dim>::assemble_matrix(VOFScratchData<dim> &      scratch_data,
       // const Tensor<2, dim> k_corr      = (r * s) * outer_product(s,
       // s);
 
-      // Calculate the unit vector associated with the phase gradient
-      Tensor<1, dim> r = phase_gradient / (phase_gradient_norm + tol);
+      Tensor<1, dim> r = phase_gradient / (phase_gradient.norm() + tolerance);
```
[the remark this answers is not in the document]
Comment (blaisb) [reply source/solvers/vof_assemblers.cc:76]: I used r because it was the notation from the article, but I'll use normalized gradient.

**Review by blaisb — APPROVED** [review]

**Pull request merged**